### PR TITLE
Initial impl of hf transformers adapter.

### PR DESCRIPTION
**Pull request by deeply**

### CODE REVIEW

**Comment by tanyokwok** [comment]
cc @zzpmiracle Hf Diffusers’ BladeDISC optimization can be aligned with hf Transformers'.

File: examples/PyTorch/Inference/hf_transformers/blade_adapter.py [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+    tracable = Tracable(model, [], info.default_kwargs, amp=amp)
+    traced = torch.jit.trace(tracable.eval(), _kwargs_to_args(
+        info.input_order, **info.dummy_inputs), strict=False)
+
+    if skip_compile:
+        opt_model = traced
+    else:
+        # TODO(litan.ls): support custom blade config
+        torch_config = torch_blade.config.Config()
+        torch_config.enable_mlir_amp = False  # disable mix-precision
```
Comment (chenbohua3) [inline examples/PyTorch/Inference/hf_transformers/blade_adapter.py:245]: The default value of `enable_mlir_amp` is False, there is no need to set it explicitly

File: examples/PyTorch/Inference/hf_transformers/blade_adapter.py [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+
+# pipeline task to model task mapping
+TASK_MAP = {
+    'audio-classification': 'sequence-classification',  # ?
+    'automatic-speech-recognition': 'speech2seq-lm',
+    'feature-extraction': 'default',  # ?
+    'text-classification': 'sequence-classification',
+    'token-classification': 'token-classification',
+    'question-answering': 'question-answering',
+    # 'table-question-answering': '',
```
Comment (qiuxiafei) [inline examples/PyTorch/Inference/hf_transformers/blade_adapter.py:58]: any problem with commented tasks?

**Comment by zzpmiracle** [comment]
> cc @zzpmiracle Hf Diffusers’ BladeDISC optimization can be aligned with hf Transformers'.

I think not. Diffusers has diffrent abstaction called pipeline, so we need to extract modules like encoder, unet..., and optimize them individually, so may be hard to do this.

**Comment by tanyokwok** [comment]
> > cc @zzpmiracle Hf Diffusers’ BladeDISC optimization can be aligned with hf Transformers'.
> 
> I think not. Diffusers has diffrent abstaction called pipeline, so we need to extract modules like encoder, unet..., and optimize them individually, so may be hard to do this.

I want to notify you about this PR only. But you are more hands-on than me. I think you are right.

File: examples/PyTorch/Inference/hf_transformers/README.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+*(under development)*
+
+## 1. Prerequisites
+
+- torch==1.12.0+cu113
+- transformers>=4.24.0
+- Latest BladeDISC (torch_blade), pull docker image `bladedisc/bladedisc:latest-runtime-torch1.12.0-cu113`
+
+Please note that, you may need to build torch_blade from source if using difference torch and cuda versions.
+Refer to [build BladeDISC for PyTorch]().
```
Comment (tanyokwok) [inline examples/PyTorch/Inference/hf_transformers/README.md:12]: Nit: Please add the link

File: examples/PyTorch/Inference/hf_transformers/blade_adapter.py (hunk 1 above)
Comment (deeply) [reply examples/PyTorch/Inference/hf_transformers/blade_adapter.py]: Removed it.

File: examples/PyTorch/Inference/hf_transformers/blade_adapter.py (hunk 2 above)
Comment (deeply) [reply examples/PyTorch/Inference/hf_transformers/blade_adapter.py]: I've not verified these pipeline tasks to model tasks mapping.  I added a TODO to support these tasks in the updated commit.

File: examples/PyTorch/Inference/hf_transformers/README.md (hunk 3 above)
Comment (deeply) [reply examples/PyTorch/Inference/hf_transformers/README.md]: Done.

**Pull request merged**